### PR TITLE
Add fuse batchnorm pass

### PR DESCRIPTION
**Pull request by Erik-Lundell**
The pass differs from existing fuse passes since they
use the get_attr node which is not supported by ArmBackend.
Instead, we update the existing parameters.
Also adds tests.

To test the batchnorm pass, I extended RunPasses, adding a parameter,
`passes_with_exported_program`, that can be used just
like `pass_list` but are initiated with an
exported program before they are run.

The functionality is tested in new tests for the
CastInt64Pass and InsertTableOpsPass

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218

### CODE REVIEW

File: backends/arm/_passes/fuse_batchnorm2d_pass.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Copyright 2025 Arm Limited and/or its affiliates.
```
Comment (digantdesai) [inline backends/arm/_passes/fuse_batchnorm2d_pass.py:1]: Sorry just skimmed, can we somehow share this pass with other delegates? I know for instance XNNPACK also has similar pass. We have `backends/transforms` directory for something like this.

Comment (Erik-Lundell) [reply backends/arm/_passes/fuse_batchnorm2d_pass.py:1]: > The pass differs from existing [batchnorm] fuse passes since they
use the get_attr node which is not supported by ArmBackend.
Instead, we update the existing parameters.

So the logic for how the conv parameters are updated is different from those passes.
I also think the existing pass in backends/transforms misses some logic, for example it doesn't handle if the batchnorm weight is None, or check that the conv node has only the batchnorm as a user. I was not sure whether I should update this, if I break things for other users.

I could move the pass to transforms/fuse_batch_norm_with_conv and try to share as much logic as possible between the two passes, only keeping the difference of how the parameters are update. WDYT?

Comment (digantdesai) [reply backends/arm/_passes/fuse_batchnorm2d_pass.py:1]: I guess my hope was to avoid duplication across backends. But if you think (like XNNPACK apperently) its more trouble updating the shared pass then well at least we tried. :)

**Review by zingo — APPROVED** [review]

**Comment by zingo** [comment]
macos unrealted

**Review by digantdesai — COMMENTED** [review]
Dang missed the train on this one :p

File: backends/arm/_passes/arm_pass_manager.py [hunk 2]
```diff
@@ -126,6 +127,7 @@ def _tosa_080_MI_pipeline(self, exported_program: ExportedProgram) -> GraphModul
         self.add_pass(ConvertMeanDimToAveragePoolPass())
         self.add_pass(DecomposeDivPass())
         self.add_pass(DecomposeSoftmaxesPass())
+        self.add_pass(FuseBatchnorm2DPass(exported_program))
```
[the remark this answers is not in the document]
Comment (Erik-Lundell) [reply backends/arm/_passes/arm_pass_manager.py:130]: This is already done in the BI case, in `prepare_pt2e` I believe

File: backends/arm/test/passes/test_cast_int64_pass.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import torch
+from executorch.backends.arm._passes.cast_int64_pass import CastInt64ToInt32Pass
+
+from executorch.backends.arm.test import common
+
+from executorch.backends.arm.test.tester.arm_tester import ArmTester, RunPasses
+
+
+class Int64Model(torch.nn.Module):
```
[the remark this answers is not in the document]
Comment (Erik-Lundell) [reply backends/arm/test/passes/test_cast_int64_pass.py:17]: A scalar always becomes int64

File: backends/arm/test/passes/test_fuse_batchnorm_pass.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+    def forward(self, x):
+        x1 = self.conv2d(x)
+        x = self.batch_norm2d(x1)  # Can't be fused since x1 has multiple users
+        x = self.relu6(x)
+        y = self.conv2d2(x1)
+        z = self.conv2d2(x)
+        a = self.batch_norm2d(
+            y
+        )  # Can't be fused since paramters of conv2d2 have multiple users.
```
[the remark this answers is not in the document]
Comment (Erik-Lundell) [reply backends/arm/test/passes/test_fuse_batchnorm_pass.py:124]: Since we update the weights of the convolution, it will produce different output. The new output will only be correct for the user that had the bn.
